### PR TITLE
citations: citation_count fix

### PR DESCRIPTION
**Pull request by tomaszgy**
* add citation_count to all records, not only those that have references

Signed-off-by: Tomasz <tomaszgy@gmail.com>

### CODE REVIEW

File: inspirehep/modules/migrator/tasks.py [hunk 1]
```diff
@@ -187,11 +187,6 @@ def add_citation_counts():
             es,
             query={
                 "_source": "references.recid",
-                "filter": {
-                    "exists": {
-                        "field": "references.recid"
-                    }
-                },
```
Comment (kaplun) [inline inspirehep/modules/migrator/tasks.py]: I don't think this is necessary to solve the bug. Actually I realize the best way would really be to exploit the `citations_lookup`.

You should rather create the `records_to_update` list via a generator independent from the input.:
```python
def get_records_to_update_generator(citation_lookup):
    for recid, citation_count in citation_lookup.iteritems():
        yield {'_op_type': 'update',
                     '_index': 'hep',
                     '_type': 'record',
                     '_id': recid,
                     'doc': {'citation_count': citation_count}
                 }

for chunk in chunker(get_records_to_update_generator(citations_lookup), CHUNK_SIZE):
    es_bulk(es, chunk)
```

With the current implementation you were risking of filling up the memory with a `records_to_update` list with 1.5M records :smile: .

**Comment by tomaszgy** [comment]
Thanks for tips. Added a generator, please check.

**Comment by tomaszgy** [comment]
Also seems to me like removal of `filter` is necessary. We need recids of all records, not only those that contain references.

**Comment by kaplun** [comment]
You have the recids of all records that you need to re-index: these recids are those in the `references.recid` :wink:

File: inspirehep/modules/migrator/tasks.py [hunk 2]
```diff
@@ -187,17 +194,15 @@ def add_citation_counts():
...
                 "size": CHUNK_SIZE
             },
             scroll=u'1m',
             index="hep",
             doc_type="record"):
 
+        # initialize the key for record in citations_lookup dict in case it doesn't exist there yet
+        citations_lookup[record['_id']] = citations_lookup.get(record['_id'], 0)
```
Comment (kaplun) [inline inspirehep/modules/migrator/tasks.py]: Here's the misunderstanding. it should not be `citations_lookup[record['_id']]`, rather:
```python 
for recid in record['references.recid']:
    citations_lookup[recid]
```

**Comment by tomaszgy** [comment]
Voila!

**Pull request merged**